ReactWindows/ReactNative.Shared/UIManager/UIManagerModule.cs

```diff
@@ -163,12 +178,32 @@ public int AddMeasuredRootView(ReactRootView rootView)
         /// Detaches a root view from the size monitoring hooks in preparation for the unmount
         /// </summary>
         /// <param name="rootView">The root view instance.</param>
-        public void DetachRootView(ReactRootView rootView)
+        public async Task DetachRootViewAsync(ReactRootView rootView)
+        {
+            // Called on main dispatcher thread
+            DispatcherHelpers.AssertOnDispatcher();
+
+            await DispatcherHelpers.CallOnDispatcher(rootView.Dispatcher, () =>
+            {
+                rootView.RemoveSizeChanged();
+                return true;
+            }, true); // allow inlining
+        }
+
+        /// <summary>
+        /// Method returning a task that completes when a root view (and a lot of dependent resources on the same dispatcher thread)
+        /// completes its cleanup.
+        /// </summary>
+        /// <param name="rootTag">The root view react tag.</param>
+        public Task GetRootViewCleanupTask(int rootTag)
```

Comment:
Why do we need to provide an API for this?

Reply:
Seems like the caller could keep track of the task.

---
In reply to: [178547780](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r178547780) [](ancestors = 178547780)

Reply:
Oh, why I made it public? By mistake, I guess.

---
In reply to: [178548003](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r178548003) [](ancestors = 178548003,178547780)

Reply:
Not sure I get the part with the "caller". The idea is that one API proceeds asynchronously by awaiting on a task that is completed by some other API. So I had to create this TaskCompletionSource based task, construct it & await in API 1, and complete it from API 2.

---
In reply to: [178548003](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r178548003) [](ancestors = 178548003,178547780)